cmd/start.go

```diff
@@ -195,3 +203,60 @@ func waitForStake(stakeMonitor chain.StakeMonitor, address string, timeout int)
 	}
 	return fmt.Errorf("timed out waiting for %s to have required minimum stake", address)
 }
+
+func initializeMetrics(
+	ctx context.Context,
+	config *config.Config,
+	netProvider net.Provider,
+	stakeMonitor chain.StakeMonitor,
+) error {
+	registry, err := metrics.Initialize(
+		config.Metrics.Port,
+	)
+	if err != nil {
+		logger.Infof("metrics are not configured")
+		return nil
+	}
+
+	logger.Infof(
+		"enabled metrics on port [%v]",
+		config.Metrics.Port,
+	)
+
+	var observationTick time.Duration
+	if tick := config.Metrics.Tick; tick != 0 {
+		observationTick = time.Duration(tick) * time.Second
+	} else {
+		observationTick = 60 * time.Second
```

Comment:
I like 60sec tick for network metrics but I don't like it for ETH connectivity. I think we should have a more conservative tick for this one to do not affect available requests counts in case someone uses 3rd party provider for Ethereum. Maybe every 10 minutes is enough?

Reply:
Also, I'd put those defaults as public constants in `metrics` package and use them from here.

Reply:
We can change it this way but 10 minutes means also a huge delay regarding alerts. Isn't eth node connectivity problems something we want to know quickly?

Second thing, do we want separate config params for both ticks?

Reply:
Really depends on the setup. The majority of stakers will use a 3rd party provider for Ethereum. I expect they will be comfortable with 10 or even 30 minutes ticks. The ones running their own Ethereum clients will probably want to have 1 minute ticks. As long as those values are configurable in toml we are fine.

Reply:
I've not looked, but if this allows alerting by predefined tick OR first failure, it seems like that'd cover most cases?

Reply:
Here we just expose metrics, alerts are an external process. A typical situation regarding this metric will be as follow:

1. `eth_connectivity` metric is checked every 10 minutes internally by the client
2. An external monitoring tool (e.g. Prometheus) will call the metrics endpoint, for example, every minute but it will record the same `eth_connectivity` value most of the time.
3. An alert (e.g. defined in Prometheus) will be raised if the metric value drops below an arbitrary threshold for a defined amount of time, for example, 1 minute.

So, because of the above, we may have a situation when we detect connectivity problems even 10 minutes after they actually occurred and we receive an alert after the time mentioned in point 3.

But, as Piotr said, it is configurable so we can adjust those values according to our needs.